opal/util/ethtool.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2016 Karol Mroz.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "opal_config.h"
+
+#include <string.h>
+#include <limits.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+#ifdef HAVE_NET_IF_H
+#include <net/if.h>
+#endif
+#ifdef HAVE_LINUX_ETHTOOL_H
+#include <linux/ethtool.h>
+#endif
+#ifdef HAVE_SYS_IOCTL_H
+#include <sys/ioctl.h>
+#endif
+#ifdef HAVE_LINUX_SOCKIOS_H
+#include <linux/sockios.h>
+#endif
+
+#include "opal/util/ethtool.h"
+#include "opal/util/if.h"
+
+#if !defined(HAVE_DECL_ETHTOOL_CMD_SPEED)
+#if !defined(HAVE_STRUCT_ETHTOOL_CMD)
+struct ethtool_cmd;
```

Comment:
@jsquyres - So I think what you're saying is, since `opal_ethtool_get_speed()` is just a simple `return 0` if `... || !defined(HAVE_STRUCT_ETHTOOL_CMD)`, then all we need is:

```C
#if !defined(HAVE_DECL_ETHTOOL_CMD_SPEED)
static inline unsigned int
ethtool_cmd_speed(const struct ethtool_cmd *ep)
{
    return (ep->speed_hi << 16) | ep->speed;
}
#endif
```
Am I parsing you correctly?